### PR TITLE
Zoltan/refactor and version bump

### PR DESCRIPTION
**Pull request by zoltanbedi**
## In this pr

- Moved all the rules to the index.js file
- Bumped every package's version to latest
- Removed [deprecated rules](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v3.0.0) from `@typescript-eslint` 
  - interface-name-prefix - migrated to new rule
  - class-name-casing - removed
- Added react hooks eslint plugin with the recommended rules

### CODE REVIEW

**Comment by ryantxu** [comment]
This looks fine to me -- but I don't have any strong opinions here... @stevenvachon ?

**Comment by stevenvachon** [comment]
Now that the file have been merged, won't this apply TypeScript rules in JavaScript files?

File: index.js [hunk 1]
```diff
@@ -1,18 +1,85 @@
-'use strict';
+"use strict";
```
Comment (stevenvachon) [inline index.js:1]: This files doesn't conform to the `.editorconfig` file.

**Comment by zoltanbedi** [comment]
> Now that the file have been merged, won't this apply TypeScript rules in JavaScript files?

No it won't. I've tried it.

File: index.js (hunk 1 above)
Comment (zoltanbedi) [reply index.js:1]: There is no .editorconfig file in this repository. I've formatted it with prettier's default config. Do you want me to change this?

Comment (stevenvachon) [reply index.js:1]: Ah, right; and https://github.com/editorconfig/editorconfig/issues/236.

I think what you have is fine, then.

**Pull request merged**